static/app/components/searchSyntax/parser.tsx

```diff
@@ -899,6 +930,14 @@ export type SearchConfig = {
    * Keys considered valid for date filter types
    */
   dateKeys: Set<string>;
+  /**
+   * Disallow free text search
+   */
+  disallowFreeText: boolean;
+  /**
+   * Disallow OR
+   */
+  disallowLogicalOr: boolean;
```

Comment:
Wondering if instead we should call this `disallowOperators: Set<BooleanOperator>` so it's not so one-off-y